src/modules/commander/HealthAndArmingChecks/checks/openDroneIDCheck.cpp

```diff
@@ -41,46 +42,56 @@ void OpenDroneIDChecks::checkAndReport(const Context &context, Report &reporter)
 		return;
 	}
 
-	NavModes affected_modes{NavModes::None};
+	// ODID module arm status topic
+	open_drone_id_arm_status_s odid_module_arm_status;
 
-	if (_param_com_arm_odid.get() == 2) {
-		// disallow arming without the Open Drone ID system
-		affected_modes = NavModes::All;
-	}
+	NavModes affected_modes{NavModes::None};
 
-	if (!context.status().open_drone_id_system_present) {
-		/* EVENT
-		 * @description
-		 * Open Drone ID system failed to report. Make sure it is setup and installed properly.
-		 *
-		 * <profile name="dev">
-		 * This check can be configured via <param>COM_ARM_ODID</param> parameter.
-		 * </profile>
-		 */
-		reporter.armingCheckFailure(affected_modes, health_component_t::open_drone_id,
-					    events::ID("check_open_drone_id_missing"),
-					    events::Log::Error, "Open Drone ID system missing");
+	if (_open_drone_id_arm_status_sub.copy(&odid_module_arm_status)) {
 
-		if (reporter.mavlink_log_pub()) {
-			mavlink_log_critical(reporter.mavlink_log_pub(), "Preflight Fail: Open Drone ID system missing");
+		if (_param_com_arm_odid.get() == 2) {
+			// disallow arming without the Open Drone ID system
+			affected_modes = NavModes::All;
```

Comment:
we are copying into a local variable (uorb 'struct') from the subscribed topic. We modify the copy and fill out the message data accordingly The failure occurs if we cannot do the copy. Check any other commander check and the same logic applies.

Reply:
Example: https://github.com/PX4/PX4-Autopilot/blob/73c77f42c807ba6ec576bb4224ece88b7df2a449/src/modules/commander/HealthAndArmingChecks/checks/offboardCheck.cpp#L44

Reply:
I didn't see any `advertise()` of this uORB topic, and it's only published if you've received an incoming `MAVLINK_MSG_ID_OPEN_DRONE_ID_ARM_STATUS`. Seems like `.copy` is not working if it was never published: https://logs.px4.io/plot_app?log=fd3f182f-4703-415c-930e-b2aad8307eb9

I could arm with COM_ARM_ODID = 2 and definitely no RemoteID connected. I could arm fine with QGC, which I should NOT be able to do.

Reply:
What RID module are you using?

Reply:
you can add a px4_info in that if clause as a sanity check, but it should still prevent arming since wether the topic info is being advertised should be independent of the rid module presence (e.g you can still copy the struct empty or not): https://github.com/PX4/PX4-Autopilot/blob/73c77f42c807ba6ec576bb4224ece88b7df2a449/src/modules/commander/HealthAndArmingChecks/checks/openDroneIDCheck.cpp#L57

Reply:
I didn't remove the RID module, I never plugged it in in the first place. I'm using a daily build from a few months ago. (around 27.11.23). I could do the same thing without QGC though, so it shouldn't matter.

If we don't pass line 50 because `open_drone_id_arm_status` was never advertised (published) in the first place then it doesn't matter how we would refuse the check in line 57 as we never enter inside.

Reply:
@ThomasRigi please try with the latest commit